packages/deployer/test/sample-project/contracts/MainProxy.sol

```diff
@@ -0,0 +1,37 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "./storage/ProxyNamespace.sol";
+
+contract MainProxy is ProxyNamespace {
+    function _fallback(address implementation) internal virtual {
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            calldatacopy(0, 0, calldatasize())
+
+            let result := delegatecall(gas(), implementation, 0, calldatasize(), 0, 0)
+
+            returndatacopy(0, 0, returndatasize())
+
+            switch result
+            case 0 {
+                revert(0, returndatasize())
+            }
+            default {
+                return(0, returndatasize())
+            }
+        }
+    }
+
+    fallback() external payable virtual {
+        _fallback(_getImplementation());
+    }
+
+    receive() external payable virtual {
```

Comment:
See https://docs.soliditylang.org/en/v0.8.0/contracts.html#receive-ether-function

Reply:
From those docs:
"If no such function exists, but a payable fallback function exists, the fallback function will be called on a plain Ether transfer."

If so, then shouldn't having a payable fallback function be enough? I.e. no need for receive and for making _fallback an external function.